cdc/cdc.cc

```diff
@@ -194,4 +195,194 @@ future<> setup(service::storage_proxy& proxy, schema_ptr s) {
     });
 }
 
+class transformer final {
+public:
+    struct map_cmp {
+        bool operator()(const std::pair<net::inet_address, unsigned int>& a,
+                        const std::pair<net::inet_address, unsigned int>& b) const {
+            if (a.first.is_ipv4()) {
+                if (b.first.is_ipv4()) {
+                    if (a.first.as_ipv4_address().ip > b.first.as_ipv4_address().ip) {
+                        return false;
+                    }
+                    return a.second < b.second;
+                } else {
+                    return true;
+                }
+            } else {
+                if (b.first.is_ipv4()) {
+                    return false;
+                } else {
+                    if (a.first.as_ipv6_address().ip > b.first.as_ipv6_address().ip) {
+                        return false;
+                    }
+                    return a.second < b.second;
+                }
+            }
+        }
+    };
+    using streams_type = std::map<std::pair<net::inet_address, unsigned int>, utils::UUID, map_cmp>;
+private:
+    schema_ptr _schema;
+    schema_ptr _log_schema;
```

Comment:
suggestion, but probably not worth the effort: `transformer` doesn't need to live across different tasks. Well yeah, currently `get_transformer` returns a `future<transformer>`, but the `transformer` is only constructed in the last return statement, so its futurization is artificial. The meat of `get_transformer` is building the streams. So:
`get_transformer` could be changed to `get_streams`, and then `transformer` could store only `const schema&` instead of `schema_ptr` (it doesn't need to manage the `schema`'s lifetime).